src/actual.js

```diff
@@ -6,21 +6,25 @@ exports.handler = function (options) {
   return function (req, res, next) {
     var originHeader = req.headers['origin']
 
-    // If either no origin was set, or the origin isn't supported, continue
-    // without setting any headers
-    if (!originHeader || !matcher(originHeader)) {
+    if (originMatcher.generic(options.origins)) {
```

Comment:
Is there a way to handle this without treating the whole block as a special case? A nested `if/else` of this size increases the complexity of the code base quite a lot and fragments related changes across multiple lines in the code base.